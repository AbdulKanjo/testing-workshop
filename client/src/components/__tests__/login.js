/*
 * This is a simple unit test for a function component.
 * These are quite easy to test generally.
 */

import React from 'react'
import {generate, render, Simulate} from 'client-test-utils'
import Login from '../login'

test('calls onSubmit with the username and password when submitted', () => {
  // Arrange
  const fakeUser = generate.loginForm()
  const handleSubmit = jest.fn()
  const {queryByTestId} = render(<Login onSubmit={handleSubmit} />)

  const usernameNode = queryByTestId('username-input')
  const passwordNode = queryByTestId('password-input')
  const formNode = queryByTestId('login-form')
  const submitButtonNode = queryByTestId('login-submit')

  // Act
  usernameNode.value = fakeUser.username
  passwordNode.value = fakeUser.password
  Simulate.submit(formNode)

  // Assert
  expect(handleSubmit).toHaveBeenCalledTimes(1)
  expect(handleSubmit).toHaveBeenCalledWith(fakeUser)
  expect(submitButtonNode.type).toBe('submit')
})

test('renders with the proper CSS', () => {
  const {root} = render(<Login />)
  expect(root).toMatchSnapshot()
})
