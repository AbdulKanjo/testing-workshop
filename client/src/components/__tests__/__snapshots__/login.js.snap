// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with the proper CSS 1`] = `
.css-0,
[data-css-0] {
  background: white;
  height: 50px;
  border: none;
  border-radius: 10px;
  box-shadow: var(--shadow);
  border-bottom: 5px;
  width: 30%;
  display: block;
  margin: 0 auto 10px auto;
  padding-left: 10px;
}

.css-0::-webkit-input-placeholder,
[data-css-0]::-webkit-input-placeholder {
  opacity: 0.5;
}

.css-0::-moz-placeholder,
[data-css-0]::-moz-placeholder {
  opacity: 0.5;
}

.css-0::-ms-input-placeholder,
[data-css-0]::-ms-input-placeholder {
  opacity: 0.5;
}

.css-0::placeholder,
[data-css-0]::placeholder {
  opacity: 0.5;
}

.css-5,
[data-css-5] {
  background: var(--green);
  padding: 10px 20px;
  display: block;
  margin: auto;
  color: white;
  border: none;
  border-radius: 10px;
  box-shadow: var(--shadow);
  cursor: pointer;
  transition: 0.5s;
  -webkit-transition: 0.5s;
  -moz-transition: 0.5s;
}

.css-5:hover,
[data-css-5]:hover {
  box-shadow: var(--shadowHover);
}

@media only screen and (max-width: 744px) {
  .css-0,
  [data-css-0] {
    width: 70%;
  }
}

<div>
  <form
    data-test="login-form"
  >
    <input
      class="css-0"
      data-test="username-input"
      name="username"
      placeholder="Username..."
    />
    <input
      class="css-0"
      data-test="password-input"
      name="password"
      placeholder="Password..."
      type="password"
    />
    <input
      class="css-5"
      data-test="login-submit"
      type="submit"
    />
  </form>
</div>
`;
